src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlRangeTranslator.cs

```diff
@@ -37,53 +40,140 @@ namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Inte
     /// <remarks>
     /// See: https://www.postgresql.org/docs/current/static/functions-range.html
     /// </remarks>
-    public class NpgsqlRangeTranslator : IMethodCallTranslator
+    public class NpgsqlRangeTranslator : IMethodCallTranslator, IMemberTranslator
     {
+        /// <summary>
+        /// The backend version to target.
+        /// </summary>
+        [CanBeNull] readonly Version _postgresVersion;
+
+        /// <summary>
+        /// Constructs a new instance of the <see cref="NpgsqlRangeTranslator"/> class.
+        /// </summary>
+        /// <param name="postgresVersion">The backend version to target.</param>
+        public NpgsqlRangeTranslator([CanBeNull] Version postgresVersion) => _postgresVersion = postgresVersion;
+
         /// <inheritdoc />
         [CanBeNull]
-        public Expression Translate(MethodCallExpression expression)
+        public Expression Translate(MethodCallExpression e)
         {
-            if (expression.Method.DeclaringType != typeof(NpgsqlRangeExtensions))
+            if (e.Method.DeclaringType != typeof(NpgsqlRangeExtensions))
                 return null;
 
-            switch (expression.Method.Name)
+            switch (e.Method.Name)
             {
-            case nameof(NpgsqlRangeExtensions.Contains):
-                return new CustomBinaryExpression(expression.Arguments[0], expression.Arguments[1], "@>", typeof(bool));
+            case nameof(NpgsqlRangeExtensions.Contains) when VersionAtLeast(9, 2):
```

Comment:
I'm not sure the range translator should actually be version-aware, and avoid to translate in pre-9.2. The entire range type doesn't exist prior to 9.2, any attempt to create a range column will fail. It seems better to let EF Core send range SQL, and have PostgreSQL fail it with a PostgreSQL error, rather than get NotSupportedException from client-evaluation.

As a general rule we try not to do a client-side check where a clean, clear database-side check already exists. In other cases we do refrain from translating version-unsupported exceptions since client evaluation works (e.g. `DateTime.Add*()`), but in this case there's no valid client evaluation.

Reply:
That's a good point. I was thinking about someone using the `NpgsqlRange<T>` as a CLR type not intended for PostgreSQL consumption....but that's not something we would encourage anyways, so throwing does sounds better.

(Note this logic applies to the new member translations where they can be client evaluated. Adding the checks to the extension methods was just some over zealous copy/pasting.)